electrolytes/solvation/solvate.py

```diff
@@ -0,0 +1,545 @@
+            break
+
+        if weight_key is not None:
+            choice = random.choices(choice_smiles, weights=choice_weights, k=1)[0]
+        else:
+            choice = random.choices(choice_smiles, k=1)[0]
+
+        choice_num_atoms = solvating_info[choice]["num_atoms"]
+        choice_charge = solvating_info[choice]["charge"]
+        
+        species_smiles.append(choice)
+        total_num_atoms += choice_num_atoms
+        total_charge += choice_charge
+
+    if len(species_smiles) == 0:
+        return None
+
+    shell, shell_charge, shell_spin = generate_solvated_mol(
+        mol, charge, spin_multiplicity, species_smiles, architector_params=architector_params
+    )
+    return shell, shell_charge, shell_spin
+
+
+def generate_random_dimers(
+    mol: Molecule | Atoms,
+    charge: int,
+    spin_multiplicity: int,
+    solvating_info: Dict[str, Dict],
+    max_atom_budget: int = 200,
+    num_selections: int = 5,
+    architector_params: Dict = dict()
+) -> List[Tuple[Atoms, int, int]]:
+    """
+    Generate (quasi)random dimers (central molecule + 1 counter ion or ion) using Architector.
+
+    Args:
+        mol (Molecule | Atoms): molecule to be solvated
+        charge (int): charge of the core molecule
+        spin_multiplicity (int): spin multiplicity of the core molecule
+        solvating_info (Dict[str, int]): Map <SMILES>:info (number of atoms, charge, etc.) for potential solvating
+            molecules
+        max_atom_budget (int): Maximum number of atoms that can be in a complex
+        max_trials (int): Maximum number of attempts adding a solvating molecule
+        architector_params (Dict): parameters for Architector solvation shell generation
+
+    Returns:
+        shell (Atoms): molecule with solvation shell
+    
+    """
+
+    # Since we're only adding one molecule, don't weight.
+    # Just make sure that the total size isn't too large
+
+    this_size = len(mol)
+    budget = max_atom_budget - this_size
+
+    # Make sure solvating molecules aren't too large and aren't of like charge with the central molecule
+    real_candidates_names = [
+        k for k, v in solvating_info.items()
+        if v["num_atoms"] <= budget
+        and charge * v["charge"] <= 0
+    ]
+
+    if len(real_candidates_names) == 0:
+        return list()
+    elif len(real_candidates_names) < num_selections:
+        choices = real_candidates_names
+    else:
+        choices = random.sample(real_candidates_names, k=num_selections)
+
+    complexes = list()
+    for candidate in choices:
+        this_complex, this_complex_charge, this_complex_spin = generate_solvated_mol(
+            mol,
+            charge,
+            spin_multiplicity,
+            [solvating_info[candidate]["smiles"]],
+            architector_params
+        )
+        complexes.append((this_complex, this_complex_charge, this_complex_spin))
+
+    return complexes
+
+
+def crawl_xyz(base_dir: Path) -> List[Path]:
```

Comment:
Why do you need this instead of just
`glob.glob('**/*.xyz', recursive=True) + glob.glob('**/*.xyz.gz', recursive=True)`
 or, if you were feeling bold,
`glob.glob('**/*.xyz*', recursive=True)`
of, if slightly less bold,
`[f for f in glob.glob('**/*.xyz*', recursive=True) if f.endswith('.xyz') or f.endswith('.xyz.gz')]`

Reply:
Certainly don't. For whatever reason, I just never think to use `glob`.